include/cobs/stream.h

```diff
@@ -6,10 +6,16 @@
 #include <stdbool.h>
 #include <version.h>
 
-#if (KERNEL_VERSION_MAJOR < 3) || ((KERNEL_VERSION_MAJOR == 3) && (KERNEL_VERSION_MINOR == 0))
+#ifdef CONFIG_ZEPHYR_NRF_MODULE
+#include <ncs_version.h>
+#endif
+
+#if KERNEL_VERSION_NUMBER < 0x30300
 #include <net/buf.h>
-#else
+#elif NCS_VERSION_NUMBER < 0x20800 && KERNEL_VERSION_NUMBER < 0x40000
```

Comment:
even if there's no `NCS_VERSION_NUMBER` defined, as long as `3.1.0 <= KERNEL_VERSION_NUMBER < 4.0.0` this header should be included.

I also don't get the reason on why NRF SDK version should be involved in this. Does the NRF SDK modify the nebuf header path?

Reply:
the nrf sdk cherry picks major changes into previous versions of zephyr. So currently, their zephyr version reports 3.7.99, but they already moved that header file because they cherry picked 4.0.0 stuff.

Reply:
I see now, this approach can make things confusing. Please add this info in the commit description

Reply:
this doesn't seem correct for the case where `NCS_VERSION_NUMBER` is not defined.
IMO, the whole thing should look like:
```
#ifdef NCS_VERSION_NUMBER
// do checks based on NCS_VERSION_NUMBER
#else
// do checks based on KERNEL_VERSION_NUMBER
#endif
```

Reply:
why not? if it's not defined it evaluates to 0 and 0 is smaller than 0x20800, effectively removing the check due to the &&

Reply:
but <zephyr/net/buf.h> is the header you want to include if kernel is <4.0.0(and also > than 3.1.0). If expression evaluates to 0 because NCS_VERSION_NUMBER is not defined, then you will include <zephyr/net_buf.h>

Reply:
you're right. I was under the wrong impression that (NCS_VERSION_NUMBER < 0x20800) evaluates to 0 if NCS_VERSION_NUMBER is not defined. But you already explained above. Sorry!